File: ItineraryBuilder-Backend/Models/Itinerary.cs


namespace ItineraryBuilder_Backend.Models
{
    public class Itinerary
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string UserId { get; set; }  
        public List<ItineraryPlace>? ItineraryPlaces { get; set; }
    }

}

File: itinerarybuilder/lib/models/itinerary.dart
import 'package:itinerarybuilder/models/itinerary_place.dart';
import 'package:json_annotation/json_annotation.dart';
part 'itinerary.g.dart';
@JsonSerializable()
class Itinerary {
  String Id;
  String Name;
  List<ItineraryPlace> itineraryPlaces;

  Itinerary({
    required this.Id,
    required this.Name,
    this.itineraryPlaces = const []
  });

    factory Itinerary.fromJson(Map<String, dynamic> json) => _$ItineraryFromJson(json);
    Map<String, dynamic> toJson() => _$ItineraryToJson(this);
}